src/test_l1_dsVideoDevice.c

```diff
@@ -881,38 +889,42 @@ void test_l1_dsVideoDevice_positive_dsGetVideoCodecInfo(void)
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     // Define variables
     int result;
-    int index = 0;
     intptr_t handle = -1;
     dsVideoCodecInfo_t codecInfo;
 
     // Step 01: Initialize video devices
     result = dsVideoDeviceInit();
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
     // Step 02: Obtain video device handle
-    result = dsGetVideoDevice(index, &handle);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-    UT_ASSERT_EQUAL((handle >= 0), true);
-
-    // Step 03: Get video codec information with dsVIDEO_CODEC_MPEGHPART2
-    result = dsGetVideoCodecInfo(handle, dsVIDEO_CODEC_MPEGHPART2, &codecInfo);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 04: Get video codec information with dsVIDEO_CODEC_MPEG4PART10
-    result = dsGetVideoCodecInfo(handle, dsVIDEO_CODEC_MPEG4PART10, &codecInfo);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 05: Get video codec information with dsVIDEO_CODEC_MPEG2
-    result = dsGetVideoCodecInfo(handle, dsVIDEO_CODEC_MPEG2, &codecInfo);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 06: Get video codec information with dsVIDEO_CODEC_MPEG2 (again to replicate the provided test case)
-    result = dsGetVideoCodecInfo(handle, dsVIDEO_CODEC_MPEG2, &codecInfo);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 07: De-initialize the video devices
+    for(int i = 0; i < gDSvideoDevice_NumVideoDevices; i++){
+        result = dsGetVideoDevice(i, &handle);
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
+        UT_ASSERT_EQUAL((handle >= 0), true);
+        
+        if(gSourceType == 1) {
+	    // Step 03: Iterate over all codecs
+            for(dsVideoCodingFormat_t codec = dsVIDEO_CODEC_MPEGHPART2; codec < dsVIDEO_CODEC_MAX; codec = (dsVideoCodingFormat_t)(codec + 1)){
+                if(!(gDSVideoDeviceConfiguration[i].SupportedVideoCodingFormats & codec)){
+                    continue;
+                }
+                result = dsGetVideoCodecInfo(handle, codec, &codecInfo);
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                // Compare with profile file
+                UT_ASSERT_EQUAL(codecInfo.num_entries, gDSVideoDeviceConfiguration[i].num_codec_entries);
+                if(codec == dsVIDEO_CODEC_MPEGHPART2){
+                    UT_ASSERT_EQUAL(codecInfo.entries->profile, gDSVideoDeviceConfiguration[i].profile);
+                    // Support for float values in KVP
+                    UT_ASSERT_EQUAL(codecInfo.entries->level, gDSVideoDeviceConfiguration[i].level);
```

Comment:
Update parse file to read the float values

Reply:
done